to check the operation

Graev opened the pull request.

Graev committed:
to check the operation

Boortcore commented:
Необходимо исправить:
- закомментировано подключение библиотеки dotenv, соответственно, информация о ключе и адресе базы данных не будут браться из файла `.env` в режиме `production`
- npm start не устанавливает NODE_ENV в значение `production`, следовательно, ключ никогда не будет браться из process.env 
![image](https://user-images.githubusercontent.com/17438227/73592471-e6f51680-450b-11ea-9ee7-b2251d9d2226.png)
- невозможно проверить работоспособность api. celebrate не даёт создать пользователя с валидной ссылкой на аватар. 
![image](https://user-images.githubusercontent.com/17438227/73592485-14da5b00-450c-11ea-9ee1-f73e13792279.png)
- неопределенная переменная next
![image](https://user-images.githubusercontent.com/17438227/73592532-8f0adf80-450c-11ea-89de-727ff1826e0e.png)

Можно лучше: 
- '/*' -  вместо такого пути лучше использовать '*'. 
- валидацию celebrate можно описать в отдельном модуле и возвращать из него переменные, которые подключаются в файле app.js. Это так же сделает корневой файл почище.

```
// requestValidation.js
export const validateData = celebrate({
  body: Joi.object().keys({
    email: Joi.string().required().min(6).max(38),
    password: Joi.string().required().min(8),
  }),
})

// app.js
import {validateData } from 'requestValidation'
app.post('/signin', validateData, login);
- роутинг неплохо бы реализовать в файле index.js папки routes, не нагружая файл app.js этой логикой. 

Резюме:
Необходимо ознакомиться с ограничением Joi.string().domain(). Оно не подойдёт для валидации url
https://github.com/hapijs/joi/blob/master/API.md#stringdomainoptions
Я продолжу проверку работы, когда появится возможность.

Graev committed:
error correction

к файлу config подключен env, в npm start dev NODE_ENV задается как develop, изменена валидация ссылок в Joi, определены переменные next, валидация перемещена в отдельный файл, роутинг перемещен в routes/index.js

Boortcore commented:
Все ошибки исправлены. 

Можно лучше:
При установке лайка лучше возвращать больше данных о карточку, а не только id лайкнувших карточку пользователей, чтобы клиент мог из данных сервера мог понять, какую карточку лайкнули/дизлайкнули.
